test/app/Makefile

```diff
@@ -58,7 +60,10 @@ APP_SRCS += $(wildcard gen/src/*.c)
 
 # Flags
 
-APP_CFLAGS += $(INC_FLAGS) -O2 -w -Wall -Werror -flto
+ACCELERATOR_UPPERCASE := $(shell echo $(ACCELERATOR) | tr [:lower:] [:upper:])
+APP_CFLAGS += -DNNX_ACCELERATOR=\"$(ACCELERATOR)\" -DNNX_$(ACCELERATOR_UPPERCASE)
+
+APP_CFLAGS += -O2 -w -Wall -Werror -flto
 APP_LDFLAGS += -flto
```

Comment:
I would remove `-flto` (which can cause sporadic problems on the RISCV GCC Toolchain in PULP Platform) and specify the c standard `-std=cXX` where XX is the version you prefer.

Reply:
I think it's not unreasonable, however, flto is a program-level option, which means that the code calling into your library would have to be compiled with flto. I suppose for a test app this isn't an issue, but it might lead to people copying this Makefile into a context where flto causes issues.

Regarding the C std, I think C99 and C11 are reasonable targets, but if you really want you can also go newer. I'd suggest NOT targeting GNU extended versions (as gcc does by default!) since those can lead to tricky compliance issues with LLVM.

Reply:
I removed the `-flto` here https://github.com/pulp-platform/pulp-nnx/pull/2/commits/8232527ca95633abcee3c59f4eb648bbac257a06
I added `-std=c11` here https://github.com/pulp-platform/pulp-nnx/pull/2/commits/df3f5ddc9d1f503e7579194a696fd731a64bf1b6 but had to check if it was compiled for GAP_SDK because they use gcc's `asm` instead of `__asm__`.
I'm not the happiest with how I check if we are compiling for GAP_SDK, so if anyone has suggestions, feel free to give them.

Reply:
Removed `-std=c11` in the end and will add it for library only when I add cmake support.